content/en/security_platform/application_security/troubleshooting.md

```diff
@@ -0,0 +1,436 @@
+
+```ruby
+Datadog.configure do |c|
+  c.diagnostics.debug = true
+end
+```
+
+Debug logs are verbose but useful. If you open up a ticket with [Datadog support][1], forward the logs with your request.
+
+#### Check #1: Is AppSec correctly enabled?
+
+Application Security has been correctly enabled if you see logs such as:
+
+```
+D, [2021-12-14T11:03:32.167125 #73127] DEBUG -- ddtrace: [ddtrace] (libddwaf/lib/datadog/security/waf.rb:296:in `block in logger=') {:level=>:ddwaf_log_info, :func=> "ddwaf_set_log_cb", :file=>"PowerWAFInterface.cpp", :message=>"Sending log messages to binding, min level trace"}
+D, [2021-12-14T11:03:32.200491 #73127] DEBUG -- ddtrace: [ddtrace] (libddwaf/lib/datadog/security/waf.rb:296:in `block in logger=') {:level=>:ddwaf_log_debug, :func= >"parse", :file=>"parser_v2.cpp", :message=>"Loaded 124 rules out of 124 available in the ruleset"}
+```
+
+If you do not see those logs, check the following:
+
+- The correct Application Security environment variables are set up for your application process. 
+- The latest gem version is installed.
+- The tracer is configured correctly and sending APM traces to your APM dashboard.
+
+#### Check #2a: Is Application Security called on HTTP requests?
+
+To confirm that Application Security is called for each HTTP request, trigger a [test attack](#1-send-a-test-attack-to-your-application) and look for these logs:
+
+```
+D, [2022-01-19T21:25:50.579745 #341792] DEBUG -- ddtrace: [ddtrace] (/home/lloeki/src/github.com/DataDog/dd-trace-rb/lib/datadog/security/reactive/operation.rb:14:in `initialize') operation: rack.request initialize
+D, [2022-01-19T21:25:50.580300 #341792] DEBUG -- ddtrace: [ddtrace] (/home/lloeki/src/github.com/DataDog/dd-trace-rb/lib/datadog/security/contrib/rack/gateway/watcher.rb:25:in `block (2 levels) in watch') root span: 964736568335365930
+D, [2022-01-19T21:25:50.580371 #341792] DEBUG -- ddtrace: [ddtrace] (/home/lloeki/src/github.com/DataDog/dd-trace-rb/lib/datadog/security/contrib/rack/gateway/watcher.rb:26:in `block (2 levels) in watch') active span: 964736568335365930
+D, [2022-01-19T21:25:50.581061 #341792] DEBUG -- ddtrace: [ddtrace] (/home/lloeki/src/github.com/DataDog/dd-trace-rb/lib/datadog/security/contrib/rack/reactive/request.rb:34:in `block in subscribe') reacted to ["request.headers", "request.uri.raw", "request.query", "request.cookies", "request.body.raw"]: [{"version"=>"HTTP/1.1", "host"=>"127.0.0.1:9292", "accept"=>"*/*", "user-agent"=>"Nessus SOAP"}, "http://127.0.0.1:9292/", [], {}, ""]
+```
+
+If you don’t see those logs, try the following:
+
+- Check that another upstream security system is not filtering requests based on the test header value, which would prevent the request from reaching the application. 
+- Send another [test attack](#1-send-a-test-attack-to-your-application) using another user's agent values in the curl command to see if the event is successfully sent.
```

Comment:
```suggestion
- Send another [test attack](send-a-test-attack-to-your-application) using another user's agent values in the curl command to see if the event is successfully sent.
```

Reply:
Updating headers to remove hashes as we typically avoid where we can. Best practice is to use H3-H5s for these instances.